custom-attachments/src/main/java/io/getstream/chat/android/customattachments/factory/DateAttachmentFactory.kt

```diff
@@ -0,0 +1,56 @@
+package io.getstream.chat.android.customattachments.factory
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.customattachments.databinding.ItemDateAttachmentBinding
+import io.getstream.chat.android.ui.message.list.adapter.MessageListListenerContainer
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.AttachmentFactory
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.InnerAttachmentViewHolder
+
+class DateAttachmentFactory : AttachmentFactory {
+
+    override fun canHandle(message: Message): Boolean {
+        return message.attachments.any { it.type == "date" }
+    }
+
+    override fun createViewHolder(
+        message: Message,
+        listeners: MessageListListenerContainer?,
+        parent: ViewGroup,
+    ): InnerAttachmentViewHolder {
+        return ItemDateAttachmentBinding
+            .inflate(LayoutInflater.from(parent.context), parent, false)
+            .let { DateAttachmentViewHolder(it, listeners) }
+    }
+
+    class DateAttachmentViewHolder(
+        private val binding: ItemDateAttachmentBinding,
+        listeners: MessageListListenerContainer?,
+    ) : InnerAttachmentViewHolder(binding.root) {
+
+        private lateinit var message: Message
+
+        init {
+            binding.dateTextView.setOnClickListener {
```

Comment:
Should we maybe put the click listeners on the containers here instead of the text view?
It's a bit of an awkward click experience.